solidity/contracts/libs/isms/MultisigIsmMetadata.sol

```diff
@@ -6,16 +6,18 @@ pragma solidity >=0.8.0;
  * [   0:  32] Merkle root
  * [  32:  36] Root index
  * [  36:  68] Origin mailbox address
- * [  68:1092] Merkle proof
- * [1092:????] Validator signatures, 65 bytes each, length == Threshold
+ * [  68:  90] Message ID (leaf at root index in merkle tree)
+ * [  90:????] Validator signatures, 65 bytes each, length == Threshold
+ * [????:????] (Optional) Merkle proof
  */
 library MultisigIsmMetadata {
     uint256 private constant MERKLE_ROOT_OFFSET = 0;
     uint256 private constant MERKLE_INDEX_OFFSET = 32;
     uint256 private constant ORIGIN_MAILBOX_OFFSET = 36;
-    uint256 private constant MERKLE_PROOF_OFFSET = 68;
-    uint256 private constant SIGNATURES_OFFSET = 1092;
+    uint256 private constant MESSAGE_ID_OFFSET = 68;
```

Comment:
Dumb question by why do we have the root in the metadata in the first place? Shouldn't we be deriving it from the proof?